cypress/integration/buy-shirt.spec.ts

```diff
@@ -1,36 +1,77 @@
+    // cy.get("button#SubmitLogin > span").click();
+    // cy.get("div#center_column button[type=\"submit\"] > span").click();
+    // cy.get("#cgv").click();
+    // cy.get("form#form button[type=\"submit\"] > span").click();
+    // cy.get("div#HOOK_PAYMENT div:nth-child(1) > div > p > a").click();
+    // cy.get("p#cart_navigation span").click();
+    // cy.get("#center_column > div > p > strong").should("have.text", "Your order on My Store is complete.");
+
+    // // Step 1. Go to http://automationpractice.com/index.php
+    menuContentPage.visitMenuContentPage();
+    // // Step 2. Go to T-Shirt
+    menuContentPage.goToTShirtMenu();
+    // // Step 3. Add to Cart first T-Shirt
+    productsListPage.addProductToCart();
+    // // Step 4. Click on "Proceed to checkout" button on pop-up window
+    // // (The one with the "Product successfully added..." message)
+    productsListPage.continueToCheckoutPopup();
     // Step 5. Click on "Proceed to checkout" on the Summary step
-    cy.get(".cart_navigation span").click();
+    shoppingCartPage.continueToCheckoutCart();
     // Step 6. Write "Email" in "Sign In" step
-    cy.get("#email").type("aperdomobo@gmail.com");
+    loginPage.enterEmailInLogin();
     // Step 7. Write "Password" in "Sign In" step
-    cy.get("#passwd").type("WorkshopProtractor");
-    // Finish missing steps from 8 to 13...
+    loginPage.enterPasswordInLogin();
+
+    // Start creating missing steps from 8 to 13 ...
+
     // Step 8. Click "Sign In" button
-    cy.get("button#SubmitLogin > span").click();
+    loginPage.continueToCheckoutBtnSignIn();
+
     // Step 9. Click on "Proceed to checkout" on the Address step
-    cy.get("div#center_column button[type=\"submit\"] > span").click();
-    // Step 10. Mark Terms & Conditions
-    cy.get("input#cgv").click();
+    addressStepPage.continueToCheckoutBtnAddress();
+
+    // Step 10. Mark Terms and Conditions Checkbox on the Shipping step
+    shippingStepPage.markTermsConditionsInAddress();
     // Step 11. Click on "Proceed to checkout" on the Shipping step
-    cy.get("form#form button[type=\"submit\"] > span").click();
+    shippingStepPage.continueToCheckoutBtnShipping();
+
     // Step 12. Click on "Pay by bank wire" on the Payment step
-    cy.get("div#HOOK_PAYMENT div:nth-child(1) > div > p > a").click();
+    paymentStepPage.selectPaymentMethod();
     // Step 13. Click on "I confirm my order" on the Payment step
-    cy.get("p#cart_navigation span").click();
-    // End creating missing steps
-    // Step 14. Verify that in "Final" Step title says "Your Order on My Store is complete"
-    cy.get("#center_column > div > p > strong").should("have.text", "Your order on My Store is complete.");
+    paymentStepPage.placeOrderButton();
+
+    // ^^^ End creating missing steps from 8 to 13 ^^^
+
+    // Step 14. Verify that in "Final" Step title says "Your Order on My Store is complete" on Payment step
+    paymentStepPage.verifyOrderConfirmationMsg();
```

Comment:
I prefer to leave the responsibility of assertions to the test classes instead of the page objects. You can do one of these two options:
- Make to this method receive the expected message, so the page object is not coupled to validate just the message that it has now.
- Change the method to return the confirmation message and perform the validation here.

Reply:
Refactored